### ISSUE TITLE
Update to use the token endpoint

### ISSUE THREAD

**Issue by anderspitman** [aaronpk/indielogin.com#121]
Seeing some weird behavior while trying to implement IndieAuth support [for obligator](https://github.com/lastlogin-io/obligator/tree/indieauth). The token request is coming back to my authorization endpoint. I've triple checked that I'm setting them to different values in my <link>s. I'm sure it's a typo somewhere, but wanted to check and make sure there's no chance it's on the IndieLogin side.

**Comment by anderspitman** [comment]
Here's an example HTML response return from the user profile:

```html
<!DOCTYPE html>
<html>
  <head>
    <link rel="indieauth-metadata" href="https://anderspitman.com/.well-known/indieauth-metadata">
    <link rel="authorization_endpoint" href="https://anderspitman.com/indieauth/auth" />
    <link rel="token_endpoint" href="https://anderspitman.com/indieauth/token" />
  </head>
  <body>
    <h1>Hi there</h1>
  </body>
</html>
```

**Comment by aaronpk** [comment]
Yeah I think this is IndieLogin.com doing the older behavior described in the first IndieAuth spec. I'll do a pass on this to update it to the latest spec this week. Sorry about that.

**Comment by anderspitman** [comment]
No worries just thought I was losing my mind for a minute. 99/100 times this is a typo somewhere in my brand new code.

**Comment by anderspitman** [comment]
Ohh I see this behavior indicates it only wants the profile and not an access token: https://indieauth.spec.indieweb.org/#profile-url-response. I actually quite like the simplicity of that.

**Comment by aaronpk** [comment]
Yeah, the only downside is it means it's not compatible with OAuth 2.0 then. The theory was that you could deploy a site with only an authorization endpoint and no token endpoint.

**Comment by janboddez** [comment]
Just ran into this too. Was previously able to use my custom [IndieAuth server](https://github.com/janboddez/laravel-indieauth) to log in to the wiki, but not the other day.

Turns out I'd stuck my authorization endpoint behind Laravel's `auth` "middleware," (unaware a remote client might ever have to "directly" access it, too). Fixed by ensuring a user's authenticated inside the controller instead, and CSRF prevention applied, only when an authorization form is being submitted.

Bit confusing the same URL is being used for both "API" and "web" (or "authenticated user") requests ... Then again, I _do_ understand that we should not design protocols around the tools we have access too (even if it'd make things quite a bit simpler, at least in this case, haha).

That said, what would it take to instead use the token endpoint? Is it a matter of simply storing also `$rels['token_endpoint']` in the provider details and then using that URL later on? Or would that break a ton of other things?

**Comment by aaronpk** [comment]
@janboddez yep that's another reason we moved this to separate the two endpoints, it's totally reasonable to use web middleware on the authorization endpoint that prevents API access
